src/jrd/ods.h

```diff
@@ -368,39 +368,39 @@ struct index_root_page
 	USHORT irt_count;				// Number of indices
 	struct irt_repeat
 	{
-	private:
-		friend struct index_root_page; // to allow offset check for private members
-		ULONG irt_root;				// page number of index root if irt_in_progress is NOT set, or
-									// highest 32 bit of transaction if irt_in_progress is set
-		ULONG irt_transaction;		// transaction in progress (lowest 32 bits)
-	public:
-		USHORT irt_desc;			// offset to key descriptions
-		UCHAR irt_keys;				// number of keys in index
-		UCHAR irt_flags;
-
-		ULONG getRoot() const;
-		void setRoot(ULONG root_page);
-
-		TraNumber getTransaction() const;
-		void setTransaction(TraNumber traNumber);
-
+		union
+		{
+			FB_UINT64 irt_transaction;	// transaction in progress
+			ULONG irt_root;				// page number of index root
+		};
```

Comment:
When it was private there was guarantee of consistent usage of `irt_root`, `irt_transaction `and flag `irt_in_progress`. 
Why make it public and open way for errors ?

Reply:
Honestly, private data members (and getters/setters) just look unnatural (to me) in ODS structures, IMHO they complicate things. That said, I agree about usage consistency. Now it's also consistent if you check for `isEmpty() / inProgress()` before using `irt_root / irt_transaction` and I thought that's enough. I can revert to private members if your disagree.

Reply:
> Honestly, private data members (and getters/setters) just look unnatural (to me) in ODS structures, IMHO they complicate things. That said, I agree about usage consistency. Now it's also consistent if you check for `isEmpty() / inProgress()` before using `irt_root / irt_transaction` and I thought that's enough. I can revert to private members if your disagree.

For me, it is good case when consistency is over complexity.

Reply:
OK, to be changed back.